packages/webview_flutter/webview_flutter_android/android/src/main/java/io/flutter/plugins/webviewflutter/CookieManagerHostApiImpl.java

```diff
@@ -30,7 +32,7 @@ public void setCookie(String url, String value) {
    * @param cookieManager The cookie manager to clear all cookies from.
    * @return Whether any cookies were removed.
    */
-  @SuppressWarnings("deprecation")
+  @SuppressWarnings({"deprecation", "RedundantSuppression"})
```

Comment:
This is the scenario where where the `javac` linter and the Android gradle linter disagreed. The `javac` linter claims this is deprecated, but the Android lint believes this does not need a suppression. 

After I did some investigation, It looks like the problem is that the `javac` linter doesn't understand the `if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP)` or `@RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)` logic. They're other instance of this throughout this plugin, and will probably show up in other plugins as well.

Potentially we should ignore the deprecation warning from the `javac` linter and allow the Android gradle linter to handle it?

@reidbaker FYI

Reply:
If the [Android lint](http://tools.android.com/tips/lint-checks) deprecation option covers everything the `javac` lint one does, but automatically handles the version checks, disabling deprecation for `javac` in all plugins sounds good to me. The fact that the javac version made us suppress everything even when we were already doing checks was really annoying.

Reply:
Ah, I tried it again in Android Studio and I think you're right. I must have got the error mixed up with another one. The lint command does still pass if only `@SuppressWarnings("deprecation")` is used. I should be able to change my local AS settings to ignore this warning.